src/app/header/header.component.ts

```diff
@@ -0,0 +1,37 @@
+import {Component, OnInit} from '@angular/core';
+import {Store} from "@ngrx/store";
+import {Observable} from "rxjs";
+import {selectToggle, selectUserName} from "./state/user/user.selectors";
+import {AppState} from "../content/state/countdown/countdown.state";
+import * as UserActions from "../header/state/user/user.actions"
+import {UserState} from "./state/user/user.state";
+
+@Component({
+  selector: 'app-header',
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.css']
+})
+export class HeaderComponent implements OnInit {
+
+  showModal$: Observable<boolean> = this.store.select(selectToggle);
+  userName$ = this.store.select(selectUserName);
+  isShown = false;
+  userName = '';
+
+  constructor(private store: Store<{countdown: AppState, user: UserState}>) {
+    this.showModal$.subscribe(res => {
```

Comment:
to avoid such constructor things ..

u can combine selectors with other selectors ...
... build selectors on top of other selectors
... put advanced logik into selectors 

```ts
const selectIsModalShown$ = createSelector(
   selectIsToggled$, 
   selectSomeThingElse$, 
   (isToggled, someThingElse) => { ... do something with it ... then return true; }
```

Reply:
too advanced to understand:)we can also take a look, so I can get a hint what exactly I can change in selectors